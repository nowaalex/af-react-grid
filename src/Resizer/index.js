import React from "react";
import PropTypes from "prop-types";
import { DraggableCore } from "react-draggable";

export const Resizer = React.memo(({
    className,
    type,
    style,
    index,
    onDrag,
    onStart,
    disabled,
    children
}) => (
    <DraggableCore onStart={onStart} onDrag={onDrag} disabled={disabled}>
        <div
            data-resizer-index={index}
            data-resizer-type={type}
            className={className}
            style={style}
            children={children}
        />
    </DraggableCore>
));

Resizer.propTypes = {
    type: PropTypes.oneOf([ "row", "col" ]),
    onDrag: PropTypes.func,
    onStart: PropTypes.func,
    index: PropTypes.number,
    disabled: PropTypes.bool,
    children: PropTypes.node,
    style: PropTypes.object,
    className: PropTypes.string
}

export default Resizer;