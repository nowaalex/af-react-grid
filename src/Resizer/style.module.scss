.resizerClass{
    composes: flexShrink0 from "flexclasses.module";  
    composes: cursorPointer from "helperclasses.module";
    flex-basis: 7px;
    background: #222;
}