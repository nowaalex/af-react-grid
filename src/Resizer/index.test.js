import React from "react";
import renderer from "react-test-renderer";
import Resizer from "../Resizer";

test( "Resizer component renders correctly", () => {
    const component = renderer.create(
        <Resizer />
    );

    let tree = component.toJSON();

    expect( tree ).toMatchSnapshot();
});