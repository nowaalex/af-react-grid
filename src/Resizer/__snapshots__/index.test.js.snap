// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resizer component renders correctly 1`] = `
<div
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "touchAction": "none",
    }
  }
/>
`;
